### PR TITLE
General Cyborg overhaul (Sandwich of buffs and nerfs)

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

This is a re-PR of @Rukofamicom's #6759 since they have no time to get this done. See original PR for contents.

## Why It's Good For The Game

See original PR.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

In original PR.

</details>

## Changelog
:cl: Rukofamicom, ivanmixo
balance: Cyborg repair is no longer a spammable action. Cyborg repairs will repeat automatically until finished or resources are exhausted with one click. Cyborgs also require assistance to repair now and are unable to use their welding tools to repair themselves.
balance: All cyborgs (including syndicate) that didn't already have a welder now start with a 20u capacity welder for repairing other cyborgs/general repairs to objects that are not themselves.
tweak: The self-repair module available to all cyborgs has been transformed into a burst healing ability.
tweak: The "weak" flashes available to robotics for building cyborgs have been reskinned into Cyborg Eyes and reduced to one charge. The robotics lab now has a few standard flashes on every map in case cyborgs get unruly.
balance: Engineering, Medical, Mining, Service and Clown Cyborgs have had their flashes replaced with a weaker version of it, arguably more suited to preventing harm; it blinds for ~7 seconds, but only paralyzes for 1 second. The standard module is now only available to the cyborg module types intended to take part in crew suppression: Peacekeeper and Standard. Syndicate cyborgs and other non station variants have retained their flashes.
balance: Cyborgs now progressively slow down as they receive damage, starting from when the first module goes offline and capping when the last module goes offline. Cyborgs with power in their cells will move at the same speed as a cyborg with all modules offline due to damage. Both penalties stack together if applicable.
balance: EMPs and Flashes now stun cyborgs for much less time: 3 seconds for strong EMP, 2 seconds for flash, 1.5 seconds for weak EMP. Laser pointers no longer stun cyborgs at all.
balance: EMPs unequip all active modules on cyborgs now.
balance: EMPs, Flashes and Laser pointers now blind cyborgs for 7 seconds, up from 2. This is a total blindness and does not have a small ring of vision directly around the player - they will be truly blind and unable to see in which direction an adjacent attacker is located.
balance: EMPs now disable radios for two minutes, cyborgs included.
balance: Cyborg hyposprays now have a short use time so that they will not function on actively moving targets.
balance: Emagged Janitor cyborg has had their acid spray upgraded to Fluorosulfuric acid, to bring it slightly closer to other emagged modules.
balance: Emagged clown cyborgs now have 3u superlube instead of 3u cfl3 in their hacked flower. Honk Honk.
balance: Emagged standard cyborgs will find that their energy sword now costs 150 cell charge to use.
balance: Emagged Engineering and Mining cyborgs will find that their stun arm now obeys the cooldown that is supposed to be present on all melee weapons, and also that it costs 300 cell charge to use. Its stamina damage was also slightly reduced to match stun batons.
balance: Emagged service borgs will find their shaker stocked with Bastion Bourbon, Zombie powder, and Smoke Powder (with an activator) instead of the now removed fake beer. Cyanide has also been removed from their hypospray's list of chems.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
See reviews on original PR

**Comment by Fronsis** [comment]
Some changes are interesting and some others are nice QoL or straight up improvement

I'm curious about 

> balance: EMPs, Flashes and Laser pointers now blind cyborgs for 7 seconds, up from 2. This is a total blindness and does not have a small ring of vision directly around the player - they will be truly blind and unable to see in which direction an adjacent attacker is located.

What other full blindness we have that last 7 seconds? Might be just be me but it seems.. a little high? I could understand it with the EMP (though the shakey-distored screen is something that i would pick though ) but the laser giving full blind is.. an interesting choice, i'm just thinking it from the PoV of it would be something ''fair'' for the borg player to see, it would feel like stun+flash (from emp, flash) would be like ''welp can't wait to die, fun 1 click win'' specially considering the new drawbacks that borgs get introduced (lower speed by damage, etc)

> balance: Engineering, Medical, Mining, Service and Clown Cyborgs have lost their Flash module. The module is now only available to the modules intended to take part in crew suppression: Peacekeeper and Standard. Syndicate cyborgs have retained their flashes.

While i understand that they... really don't need it, what are they supposed to do in a scene where they're witnessing something that goes against their core laws? (human/crew harm to give an example) not everyone knows the funny trick of buckle+*spin to do a soft-stun and since they can't hard grab you can't just try to pull the aggresor away, maybe they could have the weaker flash version that robotics use for eyes? But instead of one charge up it up to 3 so at least is enough for the borg to call for help/wait for security or move the injured player to another location

Alternative i'd like for the medborg to retain the flash.. We all know how chaotic medbay can be and there's been so many times where people has run off with diseases or has shown signs of being mentally unstable or just straight up random medbay brawl, i'm not even sure if medborg retains morphine on their hypo (or was it only on emagged?) because that would be their only chance to try to stop or help anyone but if the medborg does indeed come with morphine on their defaulthypo i can already see early lynchs because of an injection of morphine and players not letting the medborg explain anything, my opinion is that they retain a 3-charge weak flash for extreme situations where law break is happening in front of them or just let the medborg keep it because medbay is always a mess

**Comment by itsmeow** [comment]
I agree, borgs should keep their flashes. Especially with the nerfs to their speed, most borgs are now completely defenseless, and laws limit borg's ability to "validhunt" anyway, so ideally the flashes should only be used for self-defense or in the case of efficiency lawset - in that case, I can agree this change is good because it limits non-peacekeeper borgs from validhunting on efficiency, but at the cost of leaving borgs mostly defenseless on asimov which is far more common. Giving non-crew suppression borgs weaker flashes would be a good idea.

**Comment by ivanmixo** [comment]
> and laws limit borg's ability to "validhunt" anyway

If only. You see, me chain flashing this criminal to the brig is just keeping them from doing human harm.

I might just make a new flash variant that blinds them for a couple seconds, while paralyzing for about a second.

**Comment by ivanmixo** [comment]
> but the laser giving full blind is.. an interesting choice

It's not 7 seconds. It's 2.5 seconds. Removed the stun too. 

> it would feel like stun+flash (from emp, flash) would be like ''welp can't wait to die, fun 1 click win'' specially considering the new drawbacks that borgs get introduced (lower speed by damage, etc)

Everyone else has to suffer these effects, why should borgs be exempt?

**Comment by Fronsis** [comment]
> > but the laser giving full blind is.. an interesting choice
> 
> It's not 7 seconds. It's 2.5 seconds. Removed the stun too.

Fair enough, i was basing my comments on what the description of the pr said, if there's no more stun and it's only 2.5 blind seems good enough, please update the changelong to reflect that

> Everyone else has to suffer these effects, why should borgs be exempt?

I was mostly refering to the scenario of getting hit by an emp/flash alongside a 7 second blindness would be a little overkill, if the full blindness is at 2.5 then that seems balanced enough to give the borg player room to do something instead of being ided moment

**Comment by FortSumter** [comment]
I don't want the borgs to keep their flashes for the same reasons I don't like that security have them. The flashes are very powerful and are a literal war-crime to use IRL and outlawed in most countries. It doesn't makes sense in any way we would have them if realistic roleplay is really what we are aiming for.

They can keep the weaker ones as I think it's fair for them only to be able to flash 2 or 3 times before having to recharge as it allows them to still use them as a means of self defense and not so much a tool for detaining people. I do also like all the rest of these changes as well as it makes borgs easier to deal with as an antag while also making being a borg a bit more fun to play.

File: code/game/objects/items/devices/radio/radio.dm [hunk 1]
```diff
@@ -282,7 +282,7 @@
 	var/datum/signal/subspace/vocal/signal = new(src, freq, speaker, language, message, spans, message_mods)
 
 	// Independent radios, on the CentCom frequency, reach all independent radios
-	if (independent && (freq == FREQ_CENTCOM || freq == FREQ_CTF_RED || freq == FREQ_CTF_BLUE))
```
Comment (itsmeow) [inline code/game/objects/items/devices/radio/radio.dm]: bad space removals, put them back. unnecessary changes make for more merge conflicts

File: code/__DEFINES/robots.dm [hunk 2]
```diff
@@ -26,7 +26,7 @@
 #define BOT_BLOCKED			14	//!  blocked
 #define BOT_NAV				15	//!  computing navigation
 #define BOT_WAIT_FOR_NAV	16	//!  waiting for nav computation
-#define BOT_NO_ROUTE		17	//! no destination beacon found (or no route)
+#define BOT_NO_ROUTE		17	//!  no destination beacon found (or no route)
```
Comment (itsmeow) [inline code/__DEFINES/robots.dm]: only change in this file?

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 3]
```diff
@@ -312,128 +312,115 @@
...
 		to_chat(cyborg, "<span class='notice'>You activate the self-repair module.</span>")
+		working = TRUE
 		START_PROCESSING(SSobj, src)
+		playsound(cyborg.loc, activation_sound, 30)
+		update_icon()
 	else
-		to_chat(cyborg, "<span class='notice'>You deactivate the self-repair module.</span>")
-		STOP_PROCESSING(SSobj, src)
-	update_icon()
+		to_chat(cyborg, "<span class='notice'>Your self-repair module is not ready to be activated again yet.</span>")
```
Comment (itsmeow) [inline code/game/objects/items/robot/robot_upgrades.dm]: Weird phrasing. Change to "Your self-repair module is currently recharging."

File: code/game/objects/items/tools/weldingtool.dm [hunk 4]
```diff
@@ -322,10 +328,9 @@
 
 /obj/item/weldingtool/largetank/cyborg
 	name = "integrated welding tool"
-	desc = "An advanced welder designed to be used in robotic systems. Custom framework doubles the speed of welding."
+	desc = "An advanced welder designed to be used in robotic systems."
```
Comment (itsmeow) [inline code/game/objects/items/tools/weldingtool.dm]: isn't this used by engiborgs? The speed of the engiborg welder shouldn't be reduced.

File: code/modules/antagonists/devil/true_devil/_true_devil.dm [hunk 5]
```diff
@@ -106,7 +106,7 @@
 /mob/living/carbon/true_devil/assess_threat(judgment_criteria, lasercolor = "", datum/callback/weaponcheck=null)
 	return 666
 
-/mob/living/carbon/true_devil/flash_act(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0)
+/mob/living/carbon/true_devil/flash_act(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0, type, blind_duration = 25)
```
Comment (itsmeow) [inline code/modules/antagonists/devil/true_devil/_true_devil.dm]: `2.5 SECONDS`

File: code/modules/assembly/flash.dm [hunk 6]
```diff
@@ -24,7 +24,8 @@
 
 /obj/item/flashbulb/examine(mob/user)
 	. = ..()
-	. += "This one can probably just about handle [charges_left] more uses."
+	. += "[charges_left == 0 ? "The bulb is completely burnt out" : charges_left == 1 ? "This bulb will probably burn out immediately in a flash" : "This bulb can probably just about handle [charges_left] more uses."]"
```
Comment (itsmeow) [inline code/modules/assembly/flash.dm]: "This bulb will probably burn out on its next use"
Also add a period to all the options.

File: code/modules/assembly/flash.dm [hunk 7]
```diff
@@ -93,10 +99,18 @@
 	var/cooldown = 20
 	var/last_trigger = 0 //Last time it was successfully triggered.
 	var/burnt_out = FALSE
+	var/flash_duration = 25
```
Comment (itsmeow) [inline code/modules/assembly/flash.dm]: 2.5 SECONDS

File: code/modules/assembly/flash.dm [hunk 8]
```diff
@@ -93,10 +99,18 @@
...
+	var/stun_duration = 70
```
Comment (itsmeow) [inline code/modules/assembly/flash.dm]: 7 SECONDS

File: code/modules/assembly/flash.dm [hunk 9]
```diff
@@ -93,10 +99,18 @@
...
 	var/obj/item/flashbulb/bulb = /obj/item/flashbulb	//Store reference to object and run new when initialised.
 
-/obj/item/assembly/flash/handheld/weak
-	bulb = /obj/item/flashbulb/weak
+/obj/item/assembly/flash/handheld/cyborg
+	bulb = /obj/item/flashbulb/cyborg
+	name = "cyborg eye"
+	desc = "A small but powerful light with a shielded camera lens embedded in the center. Cyborgs need these to see what they're doing. The camera is non-functional until it is attached to a cyborg head"
```
Comment (itsmeow) [inline code/modules/assembly/flash.dm]: add a period

File: code/modules/assembly/flash.dm [hunk 10]
```diff
@@ -107,8 +121,7 @@
 
 /obj/item/assembly/flash/examine(mob/user)
 	. = ..()
-	. += "[bulb ? "The bulb looks like it can handle just about [bulb.charges_left] more uses.\nIt looks like you can cut out the flashbulb with a pair of wirecutters." : "The device has no bulb installed."]"
-
+	. += "[!bulb ? "The device has no bulb installed." : bulb.charges_left == 0 ? "It looks like you can cut the burnt flashbulb out with a pair of wirecutters." : bulb.charges_left == 1 ? "The bulb looks like it will burn out if used" : "The bulb looks like it can handle just about [bulb.charges_left] more uses."]"
```
Comment (itsmeow) [inline code/modules/assembly/flash.dm]: missing period on one charge condition

File: code/modules/assembly/flash.dm [hunk 11]
```diff
@@ -266,13 +279,13 @@
...
 	if(iscarbon(M))
-		flash_carbon(M, user, 5, 1)
+		flash_carbon(M, user, 1)
 		return TRUE
 	else if(issilicon(M))
 		var/mob/living/silicon/robot/R = M
 		log_combat(user, R, "flashed", src)
 		update_icon(1)
-		R.Paralyze(70)
+		R.Paralyze(20)
```
Comment (itsmeow) [inline code/modules/assembly/flash.dm]: 2 SECONDS

File: code/modules/assembly/flash.dm [hunk 12]
```diff
@@ -336,6 +349,12 @@
 /obj/item/assembly/flash/cyborg/wirecutter_act(mob/living/user, obj/item/I)
 	return
 
+/obj/item/assembly/flash/cyborg/weak
+	name = "cyborg light"
+	bulb = /obj/item/flashbulb/recharging/cyborg/weak
+	flash_duration = 70
+	stun_duration = 10
```
Comment (itsmeow) [inline code/modules/assembly/flash.dm]: use SECONDS defines

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 13]
```diff
@@ -328,7 +328,7 @@
 
 	return embeds
 
-/mob/living/carbon/flash_act(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0)
+/mob/living/carbon/flash_act(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0, type, blind_duration = 25)
```
Comment (itsmeow) [inline code/modules/mob/living/carbon/carbon_defense.dm]: 2.5 SECONDS

File: code/modules/mob/living/living_defense.dm [hunk 14]
```diff
@@ -393,10 +393,10 @@
 
 
 //called when the mob receives a bright flash
-/mob/living/proc/flash_act(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0, type = /atom/movable/screen/fullscreen/flash)
+/mob/living/proc/flash_act(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0, type = /atom/movable/screen/fullscreen/flash, blind_duration = 25)
```
Comment (itsmeow) [inline code/modules/mob/living/living_defense.dm]: SECONDS

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 15]
```diff
@@ -418,53 +418,53 @@
...
+		if(W.tool_behaviour == TOOL_WELDER || istype(W, /obj/item/stack/cable_coil))
+			user.changeNext_move(CLICK_CD_MELEE)
+			to_chat(user, "<span class='notice'>You are already busy!</span>")
+			return
+	if(W.tool_behaviour == TOOL_WELDER && (user.a_intent != INTENT_HARM))
 		user.changeNext_move(CLICK_CD_MELEE)
-		if (!getBruteLoss())
-			to_chat(user, "<span class='warning'>[src] is already in good condition!</span>")
+		if(src == user)
+			to_chat(user, "<span class='notice'>You are unable to properly repair yourself, seek assistance.</span>")
```
Comment (itsmeow) [inline code/modules/mob/living/silicon/robot/robot.dm]: "You try to reach your damaged shell with the welder, but your cyborg arms lack the versatility to repair yourself."

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 16]
```diff
@@ -418,53 +418,53 @@
...
-		if (getFireLoss() > 0 || getToxLoss() > 0)
-			if(src == user)
-				to_chat(user, "<span class='notice'>You start fixing yourself.</span>")
-				if(!do_after(user, 50, target = src))
-					return
-			if (coil.use(1))
-				adjustFireLoss(-30)
-				adjustToxLoss(-30)
+		if(!getFireLoss())
+			to_chat(user, "The wires seem fine, there's no need to fix them.")
```
Comment (itsmeow) [inline code/modules/mob/living/silicon/robot/robot.dm]: spanless to_chat

File: code/modules/mob/living/silicon/robot/robot_defense.dm [hunk 17]
```diff
@@ -82,12 +82,14 @@
 	. = ..()
 	if(. & EMP_PROTECT_SELF)
 		return
-	switch(severity)
-		if(1)
-			Stun(160)
-		if(2)
-			Stun(60)
-
+	if(severity) //catch to prevent divide by zero
```
Comment (itsmeow) [inline code/modules/mob/living/silicon/robot/robot_defense.dm]:
```
if(!severity)
    return
```

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 18]
```diff
@@ -377,7 +378,8 @@
 		/obj/item/gun/energy/disabler/cyborg,
 		/obj/item/melee/transforming/energy/sword/cyborg,
 		/obj/item/gun/energy/pulse/carbine/cyborg,
-		/obj/item/clothing/mask/gas/sechailer/cyborg)
+		/obj/item/clothing/mask/gas/sechailer/cyborg,
+		/obj/item/weldingtool/cyborg)
```
Comment (itsmeow) [inline code/modules/mob/living/silicon/robot/robot_modules.dm:383]: why does engiborg have two welders?

File: code/modules/mob/living/silicon/silicon_defense.dm [hunk 19]
```diff
@@ -140,4 +137,7 @@
 
 /mob/living/silicon/flash_act(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0, type = /atom/movable/screen/fullscreen/flash/static)
 	if(affect_silicon)
-		return ..()
+		overlay_fullscreen("flash", type)
+		addtimer(CALLBACK(src, .proc/clear_fullscreen, "flash", 25), 70) //Flashes now blind silicons for the duration they used to stun
```
Comment (itsmeow) [inline code/modules/mob/living/silicon/silicon_defense.dm]: SECONDS defines

File: code/modules/surgery/bodyparts/robot_bodyparts.dm [hunk 20]
```diff
@@ -265,16 +265,16 @@
...
 			single_flash = TRUE
-			. += "One of its eye sockets is currently occupied by a flash.\n"+\
-			"<span class='info'>It has an empty eye socket for another <b>flash</b>.</span>"
+			. += "One of its eye sockets is currently occupied by a [flash1 ? "[flash1.name]":"[flash2.name]"].\n"+\
+			"<span class='info'>It has an empty eye socket for another <b>cyborg eye</b>.</span>"
 		else
-			. += "It has two eye sockets occupied by flashes."
-		. += "<span class='notice'>You can remove the seated flash[single_flash ? "":"es"] with a <b>crowbar</b>.</span>"
+			. += "It has two eye sockets occupied by a [flash1.name] and [flash2.name]."
+		. += "<span class='notice'>You can remove the seated eye[single_flash ? "":"s"] with a <b>crowbar</b>.</span>"
```
Comment (itsmeow) [inline code/modules/surgery/bodyparts/robot_bodyparts.dm:277]: aren't there macros for this in the language? like `\plural` or something?

**Review by itsmeow — CHANGES_REQUESTED** [review]
All of the instances of 
2.5 SECONDS
and 7 SECONDS etc should be in a robot stun/paralyze DEFINE, that way the values can be easily tweaked. It seems a lot of these values are hardcoded and strewn about the several robot files.

File: code/modules/assembly/flash.dm (hunk 11 above)
Comment (itsmeow) [reply code/modules/assembly/flash.dm]: shouldn't this use the `stun_duration` var?
EDIT: add a separate var called `silicon_stun_duration`

File: code/modules/mob/living/silicon/silicon_defense.dm (hunk 19 above)
Comment (itsmeow) [reply code/modules/mob/living/silicon/silicon_defense.dm]: I hate that this just assumes the flash should be used for this length of time, is there no way to get the stun/flash duration from the caller or flash that was used?

File: code/modules/reagents/reagent_containers/borghydro.dm [hunk 21]
```diff
@@ -102,7 +102,7 @@ Borg Hypospray
 		return
 	if(!istype(M))
 		return
-	if(R.total_volume && M.can_inject(user, 1, user.zone_selected,bypass_protection))
+	if(R.total_volume && M.can_inject(user, 1, user.zone_selected,bypass_protection) && do_mob(user, M, 5))
```
Comment (itsmeow) [inline code/modules/reagents/reagent_containers/borghydro.dm]: 0.5 SECONDS

File: code/modules/reagents/reagent_containers/borghydro.dm [hunk 22]
```diff
@@ -273,7 +273,7 @@ Borg Shaker
 /obj/item/reagent_containers/borghypo/peace/hacked
 	desc = "Everything's peaceful in death!"
 	icon_state = "borghypo_s"
-	reagent_ids = list(/datum/reagent/peaceborg/inabizine,/datum/reagent/peaceborg/tire,/datum/reagent/pax/peaceborg,/datum/reagent/toxin/staminatoxin,/datum/reagent/toxin/sulfonal,/datum/reagent/toxin/sodium_thiopental,/datum/reagent/toxin/cyanide,/datum/reagent/toxin/fentanyl)
```
Comment (itsmeow) [inline code/modules/reagents/reagent_containers/borghydro.dm:276]: undocumented change

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/__DEFINES/robots.dm (hunk 2 above)
Comment (ivanmixo) [inline code/__DEFINES/robots.dm]:
```suggestion
#define BOT_NO_ROUTE		17	//! no destination beacon found (or no route)
```

File: code/modules/mob/living/silicon/robot/robot_modules.dm (hunk 18 above)
Comment (ivanmixo) [reply code/modules/mob/living/silicon/robot/robot_modules.dm:383]: image

No?

File: code/modules/surgery/bodyparts/robot_bodyparts.dm (hunk 20 above)
Comment (ivanmixo) [reply code/modules/surgery/bodyparts/robot_bodyparts.dm:277]: Not one that's applicable here. (it's `\s`)

File: code/modules/reagents/reagent_containers/borghydro.dm (hunk 22 above)
Comment (ivanmixo) [reply code/modules/reagents/reagent_containers/borghydro.dm:276]: Blame ruko

File: code/modules/mob/living/silicon/silicon_defense.dm (hunk 19 above)
Comment (ivanmixo) [reply code/modules/mob/living/silicon/silicon_defense.dm:141]: There is, but I'd need to pass it as an arg which is arguably even more messy. I'll do it if you insist, though.

File: code/game/objects/items/tools/weldingtool.dm (hunk 4 above)
Comment (ivanmixo) [reply code/game/objects/items/tools/weldingtool.dm]: This was just a pure lie from what I can tell, so.

File: code/game/objects/items/devices/radio/radio.dm (hunk 1 above)
Comment (ivanmixo) [reply code/game/objects/items/devices/radio/radio.dm]: I mean, consistency. Half the ifs have spaces between the bracket, half don't. It's annoying. I'll put it back if you insist.

File: code/game/objects/items/tools/weldingtool.dm (hunk 4 above)
Comment (itsmeow) [reply code/game/objects/items/tools/weldingtool.dm]: image

Comment (itsmeow) [reply code/game/objects/items/tools/weldingtool.dm]: Does that not do anything?

File: code/game/objects/items/devices/radio/radio.dm (hunk 1 above)
Comment (itsmeow) [reply code/game/objects/items/devices/radio/radio.dm]: I do insist, you changed three lines in the file but changed 20 others for an unenforced code standard

File: code/modules/mob/living/silicon/robot/robot_modules.dm (hunk 18 above)
Comment (itsmeow) [reply code/modules/mob/living/silicon/robot/robot_modules.dm:383]: Bleh, misread the code

**Comment by itsmeow** [comment]
It's near finished in my opinion. Just left sitting without reviews, as many PRs turn out. Feel free to mention me in discord or whatever if you want a review on something. Easy for stuff to get buried in the PR list.

File: code/game/objects/items/devices/radio/radio.dm [hunk 23]
```diff
@@ -302,7 +302,7 @@
 
 /obj/item/radio/proc/backup_transmission(datum/signal/subspace/vocal/signal)
 	var/turf/T = get_turf(src)
-	if (signal.data["done"] && (T.get_virtual_z_level() in signal.levels))
+	if ( signal.data["done"] && (T.get_virtual_z_level() in signal.levels))
```
Comment (ivanmixo) [inline code/game/objects/items/devices/radio/radio.dm]:
```suggestion
	if (signal.data["done"] && (T.get_virtual_z_level() in signal.levels))
```

File: code/game/objects/items/devices/radio/radio.dm [hunk 24]
```diff
@@ -313,16 +313,16 @@
...
 /obj/item/radio/Hear(message, atom/movable/speaker, message_language, raw_message, radio_freq, list/spans, list/message_mods = list())
 	. = ..()
-	if(radio_freq || !broadcasting || get_dist(src, speaker) > canhear_range)
+	if(radio_freq || !on || !broadcasting || get_dist(src, speaker) > canhear_range)
 		return
 
 	if(message_mods[RADIO_EXTENSION] == MODE_L_HAND || message_mods[RADIO_EXTENSION] == MODE_R_HAND)
 		// try to avoid being heard double
-		if (loc == speaker && ismob(speaker))
+		if(loc == speaker && ismob(speaker))
```
Comment (ivanmixo) [inline code/game/objects/items/devices/radio/radio.dm]:
```suggestion
		if (loc == speaker && ismob(speaker))
```

File: code/game/objects/items/devices/radio/radio.dm [hunk 25]
```diff
@@ -313,16 +313,16 @@
...
 			var/mob/M = speaker
 			var/idx = M.get_held_index_of_item(src)
 			// left hands are odd slots
-			if (idx && (idx % 2) == (message_mods[RADIO_EXTENSION] == MODE_L_HAND))
+			if(idx && (idx % 2) == (message_mods[RADIO_EXTENSION] == MODE_L_HAND))
```
Comment (ivanmixo) [inline code/game/objects/items/devices/radio/radio.dm]:
```suggestion
			if (idx && (idx % 2) == (message_mods[RADIO_EXTENSION] == MODE_L_HAND))
```

File: code/game/objects/items/devices/radio/radio.dm [hunk 26]
```diff
@@ -354,7 +354,7 @@
 
 /obj/item/radio/examine(mob/user)
 	. = ..()
-	if (frequency && in_range(src, user))
+	if( frequency && in_range(src, user))
```
Comment (ivanmixo) [inline code/game/objects/items/devices/radio/radio.dm]:
```suggestion
	if (frequency && in_range(src, user))
```

**Review by itsmeow — APPROVED** [review]
Code looks fine although I would advise just booting up the game and testing that all of the changes in the original PR are functioning as expected.

**Review by PowerfulBacon — COMMENTED** [review]
I think the flash changes should be reverted, as nerfing cyborg flashes is a poor excuse at an overall flash nerf, which we are already doing.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
My comments on the changes:
 - Cyborgs shouldnt have a unique version of the flash, if the flash needs to be nerfed then we will nerf the flash. (There is a PR for that)
 - EMPs should effect cyborgs for way longer than 3 seconds considering they are more difficult to obtain.
 - The flashes in robotics should just remain as they were before imo, rather than specifically cyborg eyes

**Comment by FortSumter** [comment]
Personally I'd like to see a day where security uses tasers and pepper spray to subdue people rather than committing literal war crimes by using flashes. As for the borg having a flash, it's somewhat okay considering that it's the only way they have of defending themselves non-lethal, though I do wish they had an alternative.

Emps are a mixed bag. They are hard to get for normal crew, but most common antags get the option to have EMP if they want. Considering they also damage the borg and can effectively chain-stun them to death, if something isn't done to make this better you may as well have EMP instakill them as that would be more realistic and keep the player from having to suffer through that shit.

**Comment by ivanmixo** [comment]
Yeah, whatever. Sorry ruko.
